fe/fe-core/src/main/java/org/apache/doris/nereids/util/JoinUtils.java

```diff
@@ -112,6 +113,7 @@ public static Pair<List<SlotReference>, List<SlotReference>> getOnClauseUsedSlot
             }
         }
 
+        Preconditions.checkArgument(childSlots.first.size() == childSlots.second.size());
```

Comment:
```suggestion
        Preconditions.checkState(childSlots.first.size() == childSlots.second.size());
```
https://guava.dev/releases/19.0/api/docs/com/google/common/base/Preconditions.html